app/models/user_course.rb

```diff
@@ -156,7 +156,7 @@ def update_exp_and_level
     self.exp = self.exp >= 0 ? self.exp : 0
 
     unless self.level
-      self.level = self.course.levels.find_by_level(0)
+      self.level = course.levels.find_by_level(0) || course.levels.first
```

Comment:
Can we just change to <code> course.levels.first </code>, because the default scope of levels is ordered by level. So <code>course.levels.first</code> actually means level 0 and it won't fail for courses who don't have level0